collect_app/src/test/java/org/odk/collect/android/preferences/AdminPreferencesActivityTest.java

```diff
@@ -0,0 +1,126 @@
+package org.odk.collect.android.preferences;
+
+import android.content.SharedPreferences;
+import android.preference.CheckBoxPreference;
+
+import org.hamcrest.Description;
+import org.hamcrest.StringDescription;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.BuildConfig;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import static org.robolectric.util.FragmentTestUtil.startFragment;
+
+/**
+ * Unit test for checking {@link SharedPreferences}'s behaviour
+ */
+@Config(constants = BuildConfig.class)
+@RunWith(RobolectricTestRunner.class)
+public class AdminPreferencesActivityTest {
+
+    private SortedMap<String, String> preferenceHashMap;
+    private AdminPreferencesFragment adminPreferencesFragment;
+    private SharedPreferences sharedPreferences;
+
+    /**
+     * Runs {@link Before} each test.
+     */
+    @Before
+    public void setUp() throws Exception {
+        adminPreferencesFragment = new AdminPreferencesFragment();
+        startFragment(adminPreferencesFragment);
+
+        sharedPreferences = adminPreferencesFragment.getActivity().
+                getSharedPreferences(AdminPreferencesActivity.ADMIN_PREFERENCES, 0);
+
+        preferenceHashMap = getPreferenceHashMap();
+    }
+
+    /**
+     * {@link Test} to assert admin setting shows and hides the associated
+     * {@link android.widget.CheckBox} in general settings
+     */
+    @Test
+    public void shouldUpdateAdminSharedPreferences() throws NullPointerException {
+        CheckBoxPreference checkBoxPreference;
+
+        for (String key : preferenceHashMap.keySet()) {
+            checkBoxPreference = (CheckBoxPreference)
+                    adminPreferencesFragment.findPreference(key);
+
+            assertPreference(checkBoxPreference, key, true);
+            assertPreference(checkBoxPreference, key, false);
+        }
+    }
+
+    /**
+     * Method to assert {@link CheckBoxPreference} with given condition
+     */
+    private void assertPreference(CheckBoxPreference cbPreference, String key, boolean condition) {
+        cbPreference.setChecked(condition);
+        boolean actual = sharedPreferences.getBoolean(preferenceHashMap.get(key), !condition);
+        if (actual != condition) {
+            assertEquals(actual, condition, preferenceHashMap.get(key));
+        }
+    }
+
+    /**
+     * Method to throw error in case expected and actual conditions didn't match
+     */
+    private void assertEquals(boolean actual, boolean expected, String value) {
```

Comment:
You should change the method name because the method is not doing what i think it's doing 👍